plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/ASTStructureValidator.xtend

```diff
@@ -378,6 +379,40 @@ class ASTStructureValidator {
 			constraints
 		)
 	}
+	
+	def private dispatch void validateASTStructure(
+		CoalesceExpression model,
+		ASTStructureDiagnosticProducer producer,
+		Set<LabelledStatement> validLabels,
+		Constraints constraints
+	) {
+		val container = model.eContainer 
+		if (container instanceof BinaryLogicalExpression) {
```

Comment:
I think an example for this case would be this: `false || (a ?? b)`? (I couldn't find a test case.)

Reply:
Both given examples do have a ParenExpression as parent. Do you want me to add test?

Reply:
Right. Could you add tests to check both of the conditions (i.e. without parentheses)?

Reply:
Done